### PR TITLE
[improve] job on yarn state patch.

### PR DESCRIPTION
**Pull request by Mrart**
## What changes were proposed in this pull request

Issue Number: close #xxx 

## Brief change log

job on yarn state patch.upate log jobmanager url

## Verifying this change

This change is already covered by existing tests, such as *(please describe tests)*.

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (no)

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkAppHttpWatcher.java [hunk 1]
```diff
@@ -575,6 +578,7 @@ and the status is not obtained this time (flink rest server is closed),
           if (trackingUrl != null && !trackingUrl.equals(application.getJobManagerUrl())) {
             application.setJobManagerUrl(trackingUrl);
             applicationService.updateJobManagerUrl(application.getId(), trackingUrl);
+            applicationLogService.updateJobManagerUrl(application.getId(), trackingUrl);
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkAppHttpWatcher.java]: Thanks for your contribution. I think the job history record should not be modified. It is just an objective record of the job. This job instance may have been stopped, so it is unreasonable to modify it.

Comment (Mrart) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkAppHttpWatcher.java]: 1. We only need to update the jobmanager url while the job is still running, not killed and cancelled states
2. The job history interface is displayed to use the jobmanagerurl, if you do not update this place to show link errors.

Comment (wolfboys) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkAppHttpWatcher.java]: > 1. We only need to update the jobmanager url while the job is still running, not killed and cancelled states
> 2. The job history interface is displayed to use the jobmanagerurl, if you do not update this place to show link errors.

In the t_flink_log table, an application (appid) may be inserted multiple times. Each operation (app build, app start, app stop, app release) will record information about this app. This PR will modify all records of the app, which will result in logical errors.

Comment (wolfboys) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkAppHttpWatcher.java]: oh, It is modified by the yarn id. Then there is no problem. Thanks.

**Comment by wolfboys** [comment]
There are some checkstyle issues here. Can you fix it? Thanks

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**